File: trunk/PO-8_210639/task_07/src/uglyfacebuilder.cpp
#include "uglyfacebuilder.h"
#include <qDebug>

void uglyFaceBuilder::buildEars() {
	ears* newEars = new ears();
	newEars->color = "red";
	newEars->figure = "square";
	Ears = newEars;
}

void uglyFaceBuilder::getResult() const
{
	qDebug() << "ugly face: ";
	Eyes->print();
	Ears->print();
	Mouth->print();
	Nose->print();
}

void uglyFaceBuilder::buildEyes() {
	eyes* newEyes = new eyes();
	newEyes->color = "black";
	newEyes->figure = "oval";
	Eyes = newEyes;
}

void uglyFaceBuilder::buildMouth() {
	mouth* newMouth = new mouth();
	newMouth->color = "red";
	newMouth->figure = "rectangle";
	Mouth = newMouth;
}

void uglyFaceBuilder::buildNose() {
	nose* newNose = new nose();
	newNose->color = "yellow";
	newNose->figure = "triangle";
	Nose = newNose;
}

File: trunk/PO-8_210648/task_07/ConsoleApp1/ConsoleApp2/UglyFaceBuilder.cs
namespace ConsoleApp2;

public class UglyFaceBuilder : Builder
{
    private Product _product = new Product();

    public override void BuildEyes()
    {
        _product.AddEyes(new string[]{"ugly","red"});
    }

    public override void BuildNose()
    {
        _product.AddNose(new string[]{"ugly","red"});
    }

    public override void BuildMouth()
    {
        _product.AddMouth(new string[]{"ugly","red"});
    }

    public override void BuildEars()
    {
        _product.AddEars(new string[]{"ugly","red"});
    }

    public override void BuildHair()
    {
        _product.AddHair(new string[]{"ugly","red"});
    }

    public override Product GetResult()
    {
        return _product;
    }
}